File: ios/TrainTrack/WatchConnectivityBridge.m
//
//  WatchConnectivityBridge.m
//  TrainTrack
//
//  React Native bridge module export
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(WatchConnectivityBridge, NSObject)

RCT_EXTERN_METHOD(sendTimerState:(NSString *)currentLabel
                  remainingSeconds:(nonnull NSNumber *)remainingSeconds
                  progress:(nonnull NSNumber *)progress
                  isCompleted:(BOOL)isCompleted)

@end

File: ios/TrainTrack/WatchConnectivityBridge.swift
//
//  WatchConnectivityBridge.swift
//  TrainTrack
//
//  React Native bridge for Watch connectivity
//

import Foundation
import React

@objc(WatchConnectivityBridge)
class WatchConnectivityBridge: NSObject {
    
    @objc
    func sendTimerState(
        _ currentLabel: String,
        remainingSeconds: NSNumber,
        progress: NSNumber,
        isCompleted: Bool
    ) {
        WatchConnectivityManager.shared.sendTimerState(
            currentLabel: currentLabel,
            remainingSeconds: remainingSeconds.intValue,
            progress: progress.doubleValue,
            isCompleted: isCompleted
        )
    }
    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }
}
